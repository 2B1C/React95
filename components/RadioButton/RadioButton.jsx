import React from 'react';
import styled from 'styled-components';

const RadioButton = styled.input.attrs({
  type: 'radio'
})`
  position: relative;
  visibility: hidden;

  &:checked {
    &:after {
      width: 6px;
      height: 6px;

      position: absolute;
      top: 3px;
      left: 3px;

      content: '';
      display: inline-block;

      background-color: #000;
      border-radius: 50%;
      visibility: visible;
    }
  }

  &:before {
    width: 12px;
    height: 12px;

    position: absolute;

    content: '';
    display: inline-block;

    background-color: #fff;
    border-radius: 50%;
    visibility: visible;
  }
`;

export default RadioButton;
