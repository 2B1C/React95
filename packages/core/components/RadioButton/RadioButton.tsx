import * as React from 'react';
import styled, { css } from '@xstyled/styled-components';
import { th } from '@xstyled/system';

import {
  radioChecked,
  radioCheckedDisabled,
  radioUnchecked,
  radioUncheckedDisabled,
} from './radioButtonStates';

export type RadioButtonProps = React.InputHTMLAttributes<HTMLInputElement>;

const Icon = styled.span`
  width: 12px;
  height: 12px;

  content: '';
  display: inline-block;

  position: absolute;
  left: 0;
  top: 0;

  background-image: url(${radioUnchecked});
`;

const Text = styled.span`
  padding: 1;
  user-select: none;

  position: absolute;
  top: 0;
  left: 18px;
`;

const Field = styled.input.attrs<RadioButtonProps>({
  type: 'radio',
})`
  margin: 0;
  opacity: 0;

  &:focus ~ ${Text}, &:active + ${Text} {
    border-width: 1;
    border-style: dotted;
    padding: 0;
  }

  &:checked + ${Icon} {
    background-image: url(${radioChecked});
  }

  &:disabled + ${Icon} {
    background-image: url(${radioUncheckedDisabled});
  }

  &:checked:disabled + ${Icon} {
    background-image: url(${radioCheckedDisabled});
  }
`;

const Label = styled.label<Pick<RadioButtonProps, 'disabled'>>`
  position: relative;
  margin-bottom: 10;
  display: block;

  ${({ disabled }) =>
    disabled &&
    css`
      color: ${th('colors.grays.3')};
      text-shadow: 0.5px 0.5px ${th('colors.grays.1')};
    `}
`;

const RadioButton: React.FC<RadioButtonProps> = ({
  children,
  onChange,
  disabled,
  ...props
}) => (
  <Label disabled={disabled}>
    <Field disabled={disabled} {...props} />
    <Icon />
    <Text>{children}</Text>
  </Label>
);

RadioButton.defaultProps = {
  children: '',
  disabled: false,
};

export default RadioButton;
