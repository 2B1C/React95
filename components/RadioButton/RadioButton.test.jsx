import React from 'react';
import RadioButton from './RadioButton';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

describe('<RadioButton />', () => {
  describe('Snapshots', () => {
    it('should match snapshot without props', () => {
      expect(renderer.create(<RadioButton />).toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with disabled prop', () => {
      expect(
        renderer.create(<RadioButton disabled />).toJSON()
      ).toMatchSnapshot();
    });
  });

  describe('onChange prop', () => {
    it('should call onChange when RadioButton is changed', () => {
      const onChangeMock = jest.fn();
      const MountedRadio = mount(
        <RadioButton onChange={onChangeMock}>Radio</RadioButton>
      );

      MountedRadio.find('input').simulate('change');

      expect(onChangeMock).toHaveBeenCalled();
    });
  });
});
