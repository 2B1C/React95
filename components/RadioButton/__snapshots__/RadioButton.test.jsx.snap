// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> Snapshots should match snapshot with disabled prop 1`] = `
.c4 {
  width: 12px;
  height: 12px;
  content: '';
  display: inline-block;
  position: absolute;
  left: 0;
  top: 0;
  background-image: url(test-file-stub);
}

.c6 {
  padding: 1px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: absolute;
  top: 0;
  left: 18px;
}

.c2 {
  margin: 0;
  opacity: 0;
}

.c2:focus ~ .c5,
.c2:active + .c5 {
  border: 1px dotted;
  padding: 0;
}

.c2:checked + .c3 {
  background-image: url(test-file-stub);
}

.c2:disabled + .c3 {
  background-image: url(test-file-stub);
}

.c2:checked .c1:disabled + .c3 {
  background-image: url(test-file-stub);
}

.c0 {
  position: relative;
  margin-bottom: 10px;
  color: #868686;
  text-shadow: 0.5px 0.5px #d2d2d2;
}

<label
  className="c0"
  disabled={true}
>
  <input
    className="c1 c2"
    disabled={true}
    onChange={[Function]}
    type="radio"
  />
  <span
    className="c3 c4"
  />
  <span
    className="c5 c6"
  >
    
  </span>
</label>
`;

exports[`<RadioButton /> Snapshots should match snapshot without props 1`] = `
.c4 {
  width: 12px;
  height: 12px;
  content: '';
  display: inline-block;
  position: absolute;
  left: 0;
  top: 0;
  background-image: url(test-file-stub);
}

.c6 {
  padding: 1px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: absolute;
  top: 0;
  left: 18px;
}

.c2 {
  margin: 0;
  opacity: 0;
}

.c2:focus ~ .c5,
.c2:active + .c5 {
  border: 1px dotted;
  padding: 0;
}

.c2:checked + .c3 {
  background-image: url(test-file-stub);
}

.c2:disabled + .c3 {
  background-image: url(test-file-stub);
}

.c2:checked .c1:disabled + .c3 {
  background-image: url(test-file-stub);
}

.c0 {
  position: relative;
  margin-bottom: 10px;
}

<label
  className="c0"
  disabled={false}
>
  <input
    className="c1 c2"
    disabled={false}
    onChange={[Function]}
    type="radio"
  />
  <span
    className="c3 c4"
  />
  <span
    className="c5 c6"
  >
    
  </span>
</label>
`;
